### PR TITLE
Feature/7753 shared exceptions for handlers

### LINKED ISSUE #7753: shared exceptions module for handlers
### Short description and motivation for the proposed feature

We should create a shared exception module for handlers, this will make error messages richer and make debugging easier

e.g [here](https://github.com/mindsdb/mindsdb/blob/d8d719753870ce05d33cff06b4ec09bb7d055c25/mindsdb/integrations/handlers/rag_handler/exceptions.py)

### Video or screenshots

_No response_

### Describe some possible solutions

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by dalalvarun**
## Description

Please include a summary of the change and the issue it solves. 

Fixes #7753 

## Type of change

(Please delete options that are not relevant)

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] 📄 This change requires a documentation update

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [ ]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Additional Media:

- [x] I have attached a brief loom video or screenshots showcasing the new functionality or change.

https://github.com/mindsdb/mindsdb/assets/70593335/0c4fbf86-5f03-4784-8592-23f71de904b8

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [ ] I have appropriately commented on my code, especially in complex areas.
- [ ] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added.

### CODE REVIEW

**Comment by dalalvarun** [comment]
Any feedback on this?

**Comment by dusvyat** [comment]
> Any feedback on this?

Thanks for your patience. Please may you fix the linting issues, see https://github.com/mindsdb/mindsdb/actions/runs/6594255068/job/17917914560?pr=7956


@ea-rus - any issues with adding this to code base?

**Review by dusvyat — CHANGES_REQUESTED** [review]
@dalalvarun - thanks for your patience. Please may you fix the linting issues, see https://github.com/mindsdb/mindsdb/actions/runs/6594255068/job/17917914560?pr=7956

**Comment by dalalvarun** [comment]
@dusvyat Sorry, I missed these linting issues. I have fixed them.

**Comment by dalalvarun** [comment]
@dusvyat Had some export path issues. My bad! Fixed them now.

File: mindsdb/exceptions.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class TableNotFound(Exception):
```
Comment (ea-rus) [inline mindsdb/exceptions.py:1]: I think this file should be located in 
mindsdb/integrations/libs, near by api_handler.py

Comment (dalalvarun) [reply mindsdb/exceptions.py]: Yea, the exceptions.py was way too common. 
Renamed it to api_handler_exceptions.py, and moved it to mindsdb\integrations\libs

**Review by dusvyat — APPROVED** [review]
thanks!

**Pull request merged**